### ISSUE TITLE
Method for access to the current prediction for both the train and test set

### ISSUE THREAD

**Issue by BlindApe** [dmlc/xgboost#84]
From the wiki:
'The buffers are used to save the prediction results of last boosting step'

Would be very interesting a method for access to the current prediction for both the train and test set.
This could speed up the xgb.iter.update function when using customized obj (the xgb.predict call each iteration) and would let to estimate micro CV error (merging cv predictions for each fold and compute the error metric based in the 'honest' join prediction).
